### DISCUSSION TITLE
Connect  to Redshift - getting db error

### DISCUSSION THREAD

**Discussion by micha-decker** [sfu-db/connector-x#302] [Q&A] [answered]
I'd like to have a fast connection to Redshift (using postgres) from my Jupyter Notebook.

**code**

```
import connectorx as cx

df = cx.read_sql("postgresql://myUser:myPass@datawarehouse-55555.eu-central-1.redshift.amazonaws.com:5439/datawarehouse-master", "select * from users", return_type="pandas")
```

**error_message**

```

`~\anaconda3\lib\site-packages\connectorx\__init__.py in read_sql(conn, query, return_type, protocol, partition_on, partition_range, partition_num, index_col)
    227             queries=queries,
    228             protocol=protocol,
--> 229             partition_query=partition_query,
    230         )
    231         df = reconstruct_pandas(result)

RuntimeError: db error: ERROR: syntax error at or near "("`
```

I think I get a connection, because if I put a table into the sql query that does not exist I get a error message like "table does not exist"

I also tried the query in my client and it works, so I conclude that the query itself is not defective.
Am I missing something?
Is there an easy way to solve this?

**Comment by wangxiaoying** [comment, marked as answer]
Hi @micha-decker , can you try to add `protocol="cursor"` when you invoke `read_sql`? By default our postgres connection uses `binary` protocol which issues the "COPY" command, and redshift does not support this. You can check more details here: https://sfu-db.github.io/connector-x/databases/postgres.html#protocols
